src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
Ok. I'll try that just now. I'll add a new commit (not repush) with the correction when finished, so that you can continue reviewing knowing nothing will change under your feet. I will reintegrate those review commits to their corresponding place when review has finished.